### DISCUSSION TITLE
Must I run MQTT

### DISCUSSION THREAD

**Discussion by gand07** [bellrichm/weewx-jas#121] [Q&A]
Hi
Do I need to run mqtt to get the website updated?

I have tried with Firefox and Brave under Windows and Safari under IOS.
The time of the last observation is not updated every 5 minutes even if I press Refresh I can see the files in Data is updated.

Gert

**Comment by bellrichm** [comment]
My guess is that the content is being cached. Dealing with cache is downfall of not using Ajax calls to retrieve the data.

**Reply by gand07** [reply]
Hi Rich
Thanks for your support.

I think you're right that it's a cache issue and I can't do anything about it?

So I will have to use mqtt to get the page updated?

Just one more question, if you look at [argon1.billeder.de](argon1.billeder.de), the time for min/max temperature is different from the graph. Min/Max is the time 16:00 and 5:40, the graph says 23:00 and 12:40. The correct time is 23:00 and 12:40

Thanks for looking

Gert

**Comment by bellrichm** [comment]
I see some ideas that might ‘fix’ the cache problem.  I’ll look into it. I’m not a nginx expert, so it may take a bit.

~~You should be able to configure the time format. I’ll look into getting that documented.~~
Oops, misread - I'll look into why the value is incorrect.

Thanks for the feedback.

**Comment by bellrichm** [comment]
This release should fix the wrong time in the min/max section.
https://github.com/bellrichm/weewx-jas/releases/tag/v1.0.1-rc01

**Comment by bellrichm** [comment]
I’m reading up on nginx configuration. Did you configure your install? Would you be willing to share it?

**Comment by gand07** [comment]
Hi
Just installed V1.0.1-rc1 and the min/max time is correct now. Thanks for that-
I can share the weewx.conf file, but I need to clean it first

Gert

**Reply by bellrichm** [reply]
I was thinking of the NGINX configuration. But I am thinking is probably cloudfare doing the caching. 
I have an idea, but unfortunately you will have to be the tester. 
Net, no configuration info is needed at this time.

**Reply by bellrichm** [reply]
Awesome on the time. Thanks for letting me know.

**Reply by bellrichm** [reply]
Oh, if you have access to the nginx configuration, you could probably turn off caching

**Reply by gand07** [reply]
Hi
I'll look at Cloudflare and the nginx cache to see if I can solve the issue.

Gert

**Reply by bellrichm** [reply]
Ok, I’m still pursuing doing sometime jas.  If you want to see if I can do something first…

**Reply by gand07** [reply]
Hi
I have changed the cache level in Cloudflare, it's bedtime in Denmark, so I'll check tomorrow 
Gert

**Reply by gand07** [reply]
BTW, I have access to the nginx server
Gert

**Reply by gand07** [reply]
I have disabled the Cloudflare cache for the host and it seems to work. I'll do some further testing and let you know
Gert

**Comment by bellrichm** [comment]
Well that sounds promising. 
I have been doing some experimenting and have an approach that seems to stop the browser from caching the data. It will take some work  to set up an environment like you have and test it. If/when I do, I will let you know the results.

**Comment by bellrichm** [comment]
I’ve added the ability to add a query string to requests for ‘data’ and ‘pages’. In Cloudfront on AWS, the query string can be included in the cache key. This makes it easy to ’break’ the cache when new data has been generated. Some additional information is here, https://github.com/bellrichm/weewx-jas/wiki/Pages#query_string_on-1
This will be in a 1.1.0 release.
I don’t think there is much else I can do, but let me know if there is.
